src/libraries/Directory.Build.targets

```diff
@@ -17,6 +14,22 @@
     <WarningsAsErrors Condition="'$(WarningsAsErrors)' == 'NU1605'" />
     <!-- Set the documentation output file globally. -->
     <DocumentationFile Condition="'$(IsSourceProject)' == 'true' and '$(DocumentationFile)' == '' and '$(MSBuildProjectExtension)' != '.vbproj'">$(OutputPath)$(MSBuildProjectName).xml</DocumentationFile>
+    <IsNETCoreAppSrc Condition="'$(IsNETCoreAppSrc)' == '' and
+                                '$(IsSourceProject)' == 'true' and
+                                $(NetCoreAppLibrary.Contains('$(AssemblyName);'))">true</IsNETCoreAppSrc>
+    <IsNETCoreAppRef Condition="'$(IsNETCoreAppRef)' == '' and
+                                '$(IsReferenceAssembly)' == 'true' and
+                                $(NetCoreAppLibrary.Contains('$(AssemblyName);')) and
```

Comment:
Why do we need the `;` at the end?

Reply:
as without the ; the contains check would match a library that isn't the exact match, ie `System.ComponentModel` in `System.ComponentModel.Composition`. Including the semicolon guarantees that we only accept exact matches.

Reply:
the only issue is when someone adds another library at the end of the `NetCoreAppLibrary` list and forgets the trailing semicolon, but that is probably rare :)

Reply:
Ah right... didn't think about that case, the reason why I was asking was the same concern as @akoeplinger mentioned.

Reply:
yeah you are right. #fixed (will push soon).

Reply:
ok undid my change as it caused problems as there was a whitespace and newline now between the last entry and the added semicolon. So I'm fine with the current approach as CI just proved that tests will fail otherwise to build as a default reference is then missing.